nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/compat/CompatSparseToDense.java

```diff
@@ -37,10 +45,30 @@ public CompatSparseToDense(INDArray indices, INDArray shape, INDArray values) {
         inputArguments.add(values);
     }
 
-    public CompatSparseToDense(INDArray indices, INDArray shape, INDArray values, INDArray defaultVaule) {
-        this(indices, shape, values);
-        Preconditions.checkArgument(defaultVaule.dataType() == values.dataType(), "Values array must have the same data type as defaultValue array");
-        inputArguments.add(defaultVaule);
+
+
+    public CompatSparseToDense(SameDiff sd, SDVariable indices, SDVariable shape, SDVariable values) {
+        super(sd,new SDVariable[]{indices,shape,values});
+    }
+
+    public CompatSparseToDense(SameDiff sd, SDVariable indices, SDVariable shape, SDVariable values, SDVariable defaultValue) {
+        super(sd,new SDVariable[]{indices,shape,values,defaultValue});
+    }
+
+    public CompatSparseToDense(INDArray indices, INDArray shape, INDArray values, INDArray defaultValue) {
+        super(new INDArray[]{indices,shape,values,defaultValue},null);
+    }
+
+    @Override
+    public List<LongShapeDescriptor> calculateOutputShape(OpContext oc) {
```

Comment:
@treo I overrode this due to silent failures for the shape compute. I highlighted this to @quickwritereader already.